src/analyzer/protocol/ssl/spicy/SSL.spicy

```diff
@@ -0,0 +1,1901 @@
+    self.parsed_version = UNKNOWN_VERSION;
+    self.flipped = False;
+    self.flip_already_alerted = False;
+    self.server_encrypted = False;
+    self.client_encrypted = False;
+    self.both_sides_encrypted_first_time = False;
+    self.established = False;
+    self.client_certificate_depth = 0;
+    self.server_certificate_depth = 0;
+  }
+};
+
+function get_encrypted(sh: Share) : bool {
+  if ( get_direction(sh) )
+    return sh.client_encrypted;
+  else
+    return sh.server_encrypted;
+}
+
+function startEncryption(handshakesink: sink&, alertsink: sink&, inout sh: Share) {
+  local old_state: bool;
+
+  if ( get_direction(sh) ) {
+    old_state = sh.client_encrypted;
+    sh.client_encrypted = True;
+  } else {
+    old_state = sh.server_encrypted;
+    sh.server_encrypted = True;
+  }
+
+  # let's see if it was not encrypted in the past....
+  if ( old_state == False ) {
+    # disconnect the sinks here? probably yes...
+    # print "Closing sink";
+    handshakesink.close();
+    alertsink.close();
+
+    if ( sh.client_encrypted && sh.server_encrypted ) {
+      # print "Encrypted first time";
+      sh.both_sides_encrypted_first_time = True;
+      sh.established = True;
+    }
+  }
+}
+
+# This function is called several times in certain circumstances.
+# If it is called twice, it is first called due to the outer version in
+# the server hello - and then again due to the supported_versions
+# field in the server hello.
+function set_version(version: uint16, inout sh: Share) : bool {
+  sh.parsed_version = version;
+  if ( version == TLSv13 || version/0xFF == 0x7F )
+    sh.tls_13 = True;
+
+  return True;
+}
+
+# public type TLSMessage = unit {
+#  %context = Share;
+#
+#  m: Message(False);
+# };
+#
+# public type DTLSMessage = unit {
+#  %context = Share;
+#
+#  m: Message(True);
+# };
+
+public type Message = unit {
+  %context = Share;
+
+  sink handshakesink;
+  sink alertsink;
+  var record_version: uint16;
+  var dtls: bool = False;
+
+  on %init {
+    self.handshakesink.connect(new Handshake(self, self.context()));
+    self.alertsink.connect(new Alert(self.context()));
+    # print "top-level init";
+  }
+
+  fragment : RecordFragmentChoice(self.handshakesink, self.alertsink, self, self.context())[];
+  # : bytes &eod if ( self.context().skipping );
+};
+
+function is_dtls_version(version: uint16) : bool {
+  if ( version == DTLSv10 || version == DTLSv12 )
+    return True;
+
+  return False;
+}
+
+type RecordFragmentChoice = unit(handshakesink: sink&, alertsink: sink&, inout msg: Message, inout sh: Share) {
+  content_type: uint8; # &convert=ContentType($$);
+  version: uint16;
+
+  switch ( is_dtls_version(self.version) ) {
+    True -> dtlsfragment : DTLSRecordFragment(self.content_type, handshakesink, alertsink, msg, sh);
+    False -> tlsfragment : TLSRecordFragment(self.content_type, handshakesink, alertsink, msg, sh);
+  };
+
+  on content_type {
+    # print "Content type", self.content_type;
+  }
```

Comment:
This hook will not magically disappear but still be invoked (and the code for that will also be generated and compiled). If you want to keep the commented out code please comment out the full hook.

Here and elsewhere.